src/docfx/lib/json/SourceInfoJsonConverter.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace Microsoft.Docs.Build
+{
+    internal class SourceInfoJsonConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType.IsGenericType && objectType.GetGenericTypeDefinition() == typeof(SourceInfo<>);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var range = JsonUtility.ToRange((IJsonLineInfo)reader);
+            var valueType = objectType.GenericTypeArguments[0];
```

Comment:
same here, unless you make SourceInfo sealed

Reply:
yeah, it could be sealed, I do not want inheritance at all on this.